edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/ModuleIdentityLifecycleManager.cs

```diff
@@ -28,6 +28,18 @@ public ModuleIdentityLifecycleManager(IIdentityManager identityManager, ModuleId
 
         public async Task<IImmutableDictionary<string, IModuleIdentity>> GetModuleIdentitiesAsync(ModuleSet desired, ModuleSet current)
         {
+            IImmutableDictionary<string, Identity> identities = (await this.identityManager.GetIdentities()).ToImmutableDictionary(i => i.ModuleId);
+
+            // Need to remove any identities (except EA/EH) that are managed by EA but don't have a tracked module for in the ModuleSet.
+            IEnumerable<string> removeCurrentIdentities = identities.Where(
+                i => !(Constants.EdgeAgentModuleIdentityName.Equals(i.Key, StringComparison.Ordinal) || Constants.EdgeHubModuleIdentityName.Equals(i.Key, StringComparison.Ordinal))).Where(
+                i => Constants.ModuleIdentityEdgeManagedByValue.Equals(i.Value.ManagedBy, StringComparison.OrdinalIgnoreCase) &&
+                     !current.Modules.Any(m => ModuleIdentityHelper.GetModuleIdentityName(m.Key) == i.Key))
+                .Select(i => i.Key);
+
+            // First any identities that don't have running modules currently.
+            await Task.WhenAll(removeCurrentIdentities.Select(i => this.identityManager.DeleteIdentityAsync(i)));
```

Comment:
Nit: can you pull this out into a separate method: RemoveStaleIdentities(identities, desired, current) or something like that? 
That way we can keep the whole code behind a flag and test it separately as well.

Reply:
Will do, also do we have an example of a feature flag somewhere I can copy paste?